### PR TITLE
Callout admin summary page

### PR DESCRIPTION
**Pull request by wpf500**
image

### CODE REVIEW

**Comment by wpf500** [comment]
I've made some progress on the summary page but have a question for you @beabee-communityrm/design.

I'd like your opinion on the position of the action buttons. Currently on large monitors they get a bit lost on the right, on smaller devices I've made them go above the rest of the page (see screenshots). I think the position of the buttons on smaller devices would work better for all screen widths. What do you think?

In general I think we could do with coming up with a clearly defined grid. Currently lots of our pages use different grids and so have different column widths etc and some pages look strange on large monitors because we have no defined max width. Can we schedule a call about this?

image

image

**Comment by rlafuente** [comment]
Cheers!

Using the mobile layout for the action buttons is a good way to unify the looks and solve the problem, but we end up missing out on the ample width of desktops. What do you think about keeping the buttons on a left-aligned column next to the main content column? Like this:

image

We'll follow up on Slack regarding the call

**Comment by wpf500** [comment]
Happy to go with this method but we need to define what the maximum width for the column is, let's discuss later!

File: src/modules/callouts/pages/CalloutsAdminPage.vue [hunk 1]
```diff
@@ -32,12 +32,12 @@
           <AppItemStatus :status="value" />
         </template>
         <template #title="{ item, value }">
-          <a
-            :href="'/tools/polls/' + item.slug"
+          <router-link
+            :to="'/admin/callouts/' + item.slug"
```
Comment (basilesimon) [inline src/modules/callouts/pages/CalloutsAdminPage.vue:36]: Also needs to change [here](https://github.com/beabee-communityrm/beabee-frontend/blob/94239ec714f16b77026da310a00929af1d89d5ff/src/modules/callouts/pages/CalloutsAdminPage.vue#L8-L10)

File: src/modules/callouts/callouts.route.ts [hunk 2]
```diff
@@ -30,4 +30,33 @@ export const calloutsRoute: Array<RouteRecordRaw> = [
       role: 'admin',
     },
   },
+  {
+    path: '/admin/callouts/:id',
+    name: 'calloutAdmin',
+    component: () => import('./pages/CalloutAdminPage.vue'),
```
Comment (basilesimon) [inline src/modules/callouts/callouts.route.ts:36]: Just on the topic of _naming things_, and from my sleep-deprived perspective:

Could this routing between be an opportunity to name pages differently, e.g.

* `/admin/callouts        =>   ./pages/TheCalloutsAdminPage.vue`
* `/admin/callouts/:id   =>   ./pages/CalloutAdminPage.vue or EditCalloutPage.vue`

File: src/modules/callouts/pages/CalloutAdminOverviewPage.vue [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+              {{ formatDistanceLocale(new Date(), callout.starts) }})</span
+            >
+          </span>
+          <span v-else>–</span>
+        </AppInfoListItem>
+        <AppInfoListItem
+          :name="t('calloutAdminOverview.dates.ends')"
+          :value="callout.expires && formatLocale(callout.expires, 'PPP')"
+        />
+      </AppInfoList>
```
Comment (basilesimon) [inline src/modules/callouts/pages/CalloutAdminOverviewPage.vue:39]: Sweet

File: src/modules/callouts/pages/CalloutAdminResponsesPage.vue [hunk 4]
```diff
@@ -0,0 +1 @@
+<template>here</template>
```
Comment (basilesimon) [inline src/modules/callouts/pages/CalloutAdminResponsesPage.vue:1]: Strong work

**Comment by basilesimon** [comment]
Amazing, thanks for putting this together so quickly 🙌

I added a couple of comments above but found nothing jarring. The only thing I'd like to emphasise is the naming which, while it makes sense, takes a second to get used to. If I understand this right, it's something like this:

```
/callouts                     → CalloutsPage (list of callouts, for users)
/callouts/:id                 → CalloutPage (single callout, for users)

/admin/callouts                → CalloutsAdminPage
/admin/callouts/:id            → CalloutAdminPage
                 |→ /          → CalloutAdminOverviewPage
                 |→ /responses → CalloutAdminResponsesPage
```

Which brings us back to something we discussed early on: how we've organised things on the filesystem v. on the router.

**Review by rlafuente — APPROVED** [review]
Super!

**Pull request merged**